pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -395,7 +395,7 @@ func getTPMapMatchingExistingAntiAffinity(pod *v1.Pod, nodeInfoMap map[string]*s
 		nodeInfo := nodeInfoMap[allNodeNames[i]]
 		node := nodeInfo.Node()
 		if node == nil {
-			catchError(fmt.Errorf("node not found"))
+			klog.Errorf("Node %s not found for pod %s", allNodeNames[i], pod.Name)
```

Comment:
From the code's perspective, `nodeInfoMap` is from cache snapshot, and if original `nodeInfo.node` is nil, it won't be snapshotted:

https://github.com/kubernetes/kubernetes/blob/9138fa58054e2115119f8a39617646894ac34c85/pkg/scheduler/internal/cache/cache.go#L230

So the error seems not to be expected in most cases. Did you meet the issue in a real env? or it's hypothetical?